core/server/common/src/main/java/alluxio/cli/validation/SecureHdfsValidationTask.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.validation;
+
+import alluxio.PropertyKey;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Task for validating security configurations.
+ */
+public class SecureHdfsValidationTask implements ValidationTask {
+  private static final String LINE_SEPARATOR = System.getProperty("line.separator").toString();
+  private static final String ALLUXIO_BIN = "./bin/alluxio";
+  private static final String GETCONF_CMD = "getConf";
+  private static final Pattern PRINCIPAL_PATTERN =
```

Comment:
Can you add some comment/exmaple to explain the pattern you are trying to match?

Reply:
Done.